api/system_profile.py

```
@@ -197,17 +196,27 @@ def get_sap_sids(search=None, tags=None, page=None, per_page=None, staleness=Non
 @api_operation
 @rbac(Permission.READ)
 @metrics.schema_validation_time.time()
-def validate_schema(repo_fork="RedHatInsights", repo_branch="master", days=1):
+def validate_schema(repo_fork="RedHatInsights", repo_branch="master", days=1, max_messages=10000):
+    # Use the identity header to make sure the user is someone from our team.
     config = Config(RuntimeEnvironment.SERVICE)
+    if get_current_identity().user.get("username") not in config.sp_authorized_users:
```

Might be good to check the presence of `user` in the identity object. Otherwise, a request with cert-auth header will fail instead of returning 403.

---

Good call. That's a quick fix